scripts/publish_site.sh

```diff
@@ -143,6 +143,9 @@ if [[ $VERSION == false ]]; then
   # Push changes to gh-pages
   cd botorch-gh-pages || exit
   git add .
+  # Exit gracefully if there's nothing to commit.
+  [[ -z $(git diff --cached --exit-code) ]] && exit 0
```

Comment:
I'm not sure this is correct bash, but it would be good to print something letting us know it intentionally exited early

```suggestion
  [[ -z $(git diff --cached --exit-code) ]] && (echo "Nothing to commit"; exit 0)
```